cns/restserver/api.go

```diff
@@ -899,6 +902,94 @@ func (service *HTTPRestService) getNetworkContainerByOrchestratorContext(w http.
 	logger.Response(service.Name, getNetworkContainerResponse, returnCode, err)
 }
 
+// getOrRefreshNetworkContainers is to check whether refresh association is needed.
+// If received  "GET": return all NCs in CNS's state file to DNC for follow-up check
+// If received "POST": Store all the NCs (from the request body that client sent) into CNS's state file
+func (service *HTTPRestService) getOrRefreshNetworkContainers(w http.ResponseWriter, r *http.Request) {
+	logger.Printf("[Azure CNS] getOrRefreshNetworkContainers")
+
+	switch r.Method {
+	case "GET":
+		// return all NCs in CNS
+		networkContainersDetails := make(map[string]cns.ContainerStatus)
+		for ncID := range service.state.ContainerStatus {
+			networkContainersDetails[ncID] = storeNcStatus(service.state.ContainerStatus[ncID])
+		}
+
+		response := cns.GetOrRefreshNetworkContainersResponse{
+			NetworkContainersDetails: networkContainersDetails,
+			Response: cns.Response{
+				ReturnCode: types.Success,
+			},
+		}
+		err := service.Listener.Encode(w, &response)
+		logger.Response(service.Name, response, response.Response.ReturnCode, err)
+
+	case "POST":
+		var req cns.GetOrRefreshNetworkContainersRequest
+		err := service.Listener.Decode(w, r, &req)
+		logger.Request(service.Name, &req, err)
+		if err != nil {
+			logger.Printf("[Azure CNS] getOrRefreshNetworkContainers failed with error: %s", err.Error())
+			return
```

Comment:
am I missing something, or do these error cases just not return anything to the client?

Reply:
Addressed.